common/src/main/java/org/apache/celeborn/common/network/server/BufferStreamManager.java

```diff
@@ -319,28 +329,71 @@ public synchronized void setupDataPartitionReader(
       streamReaders.put(streamId, dataPartitionReader);
```

Comment:
streamReaders can replace readers or readers need skip or remove when the reader is already released.

Reply:
I think this change can be done in a refactor pull request not this one.